src/SprykerSdk/Zed/ComposerConstrainer/Business/SprykerReflector/SprykerClassReflector.php

```diff
@@ -0,0 +1,310 @@
+        'ClientInterface\.php',
+        'FacadeInterface\.php',
+        'QueryContainerInterface\.php',
+        'PluginInterface\.php',
+    ];
+
+    /**
+     * @var string[]
+     */
+    protected $configurationClassSuffixes = [
+        'Config\.php',
+        'DependencyProvider\.php',
+    ];
+
+    /**
+     * @var bool
+     */
+    protected $isPublicApi;
+
+    /**
+     * @var bool
+     */
+    protected $isConfiguration;
+
+    /**
+     * @var bool
+     */
+    protected $isFactory;
+
+    /**
+     * @var bool
+     */
+    protected $isInterface;
+
+    /**
+     * @var string
+     */
+    protected $namespace;
+
+    /**
+     * @var string
+     */
+    protected $moduleName;
+
+    /**
+     * @var string
+     */
+    protected $fileName;
+
+    /**
+     * @var string
+     */
+    protected $fileContent;
+
+    /**
+     * @var string
+     */
+    protected $className;
+
+    /**
+     * @var \ReflectionClass
+     */
+    protected $reflectionClass;
+
+    /**
+     * @var bool
+     */
+    protected $isParentCore;
+
+    /**
+     * @var string
+     */
+    protected $parentNamespace;
+
+    /**
+     * @var string
+     */
+    protected $parentOrganisation;
+
+    /**
+     * @var string
+     */
+    protected $parentModuleName;
+
+    /**
+     * @var string
+     */
+    protected $parentClassName;
+
+    /**
+     * @var string
+     */
+    protected $parentPackageName;
+
+    /**
+     * @var array|null
+     */
+    protected $usedCorePackageNamesBuffer;
+
+    /**
+     * @param \SprykerSdk\Zed\ComposerConstrainer\ComposerConstrainerConfig $config
+     * @param \Symfony\Component\Finder\SplFileInfo $splFileInfo
+     */
+    public function __construct(ComposerConstrainerConfig $config, SplFileInfo $splFileInfo)
+    {
+        $this->config = $config;
+
+        $this->fileContent = $splFileInfo->getContents();
+        preg_match_all('#\nnamespace +(\\w+\\\\\\w+\\\\(\\w+)[^;]*);#', $this->fileContent, $match);
+
+        $this->moduleName = $match[2][0];
+        $this->namespace = $match[1][0];
+
+        preg_match_all('#\n(class|abstract class|interface|trait) +(\\w+)#', $this->fileContent, $match);
+        $this->className = $match[2][0];
+        $this->fileName = $splFileInfo->getFilename();
+
+        $this->isPublicApi = (bool)preg_match('#(' . implode('|', array_merge($this->publicApiClassSuffixes, $this->publicApiInterfaceSuffixes)) . ')#', $this->fileName);
+        $this->isConfiguration = (bool)preg_match('#(' . implode('|', $this->configurationClassSuffixes) . ')#', $this->fileName);
+        $this->isFactory = (bool)preg_match('#Factory\.php#', $this->fileName);
+
+        $this->reflectionClass = new ReflectionClass($this->namespace . '\\' . $this->className);
+        $this->isInterface = $this->reflectionClass->isInterface();
+
+        $this->parentClassName = $this->reflectionClass->getParentClass() ? $this->reflectionClass->getParentClass()->getShortName() : '';
```

Comment:
These are PHPStan false positives
But assigning first to a local var should solve this.